alma/controllers/admin/AdminAlmaRefunds.php

```diff
@@ -109,35 +78,26 @@ public function ajaxProcessRefund()
             $this->ajaxFail(
                 $this->module->l('There was an error while processing the refund', 'AdminAlmaRefunds')
             );
-        } else {
-            $fees = $refundResult->customer_fee;
-            $totalOrder = $refundResult->purchase_amount + $fees;
-            $totalOrderAmount = almaFormatPrice($totalOrder, (int) $order->id_currency);
-            foreach ($refundResult->refunds as $refund) {
-                $totalRefund += $refund->amount;
-            }
-            $totalRefundAmount = almaFormatPrice($totalRefund, (int) $order->id_currency);
-            $percentRefund = (100 / $totalOrder) * $totalRefund;
         }
+        $intTotalOrder = $refundResult->purchase_amount;
+        $idCurrency = (int) $order->id_currency;
+        $totalOrderAmount = almaFormatPrice($intTotalOrder, $idCurrency);
+        $intTotalRefund = $this->buildTotalRefund($refundResult->refunds, $intTotalOrder);
+        $totalRefundAmount = almaFormatPrice($intTotalRefund, $idCurrency);
+        $percentRefund = almaCalculatePercentage($intTotalRefund, $intTotalOrder);
 
         if ($isTotal) {
-            $orders = Order::getByReference($order->reference);
-            foreach ($orders as $o) {
-                $current_order_state = $o->getCurrentOrderState();
-                if ($current_order_state->id !== (int) Configuration::get('PS_OS_REFUND')) {
-                    $o->setCurrentState(Configuration::get('PS_OS_REFUND'));
-                }
-            }
+            $this->setOrdersAsRefund($order);
         }
 
         $jsonReturn = [
             'success' => true,
             'message' => $this->module->l('Refund has been processed', 'AdminAlmaRefunds'),
             'paymentData' => $refundResult,
             'percentRefund' => $percentRefund,
-            'totalRefund' => $totalRefund,
+            'totalRefund' => $intTotalRefund,
```

Comment:
if, everywhere in your code, you don't prefix your variable by the type, this is maybe not the good strategy to call this variable with `int` prefix (the same for `intTotalOrder`)

Maybe you can make difference between your variable with following suffixes:
* price => a string with separators for cents and thousand (`.`, `,`, ` `) and currency (`€`)
* amount => a number as int or float depending the context ... for example: `$totalOrderAmount`

So in your code,
* `$totalOrderAmount` should be called `$totalOrderPrice`
* `$intTotalOrder` should be called `$totalOrderAmount`

I don't know if there is important impacts with this refactoring because the json return will probably caught somewhere and used asis... but I think variables are clearly misnamed and, if we don't do that in this PR, we need to create a debt ticket to change this soon

Reply:
@syjust-alma I rename all it's better :)